packages/amplication-data-service-generator/src/util/message-broker.ts

```diff
@@ -0,0 +1,37 @@
+import { DSGResourceData, ServiceTopics } from "@amplication/code-gen-types";
+import { EnumResourceType } from "../models";
+
+export function fromServiceTopicsToTopicsWithName(
+  serviceTopicsList: ServiceTopics[],
+  messageBrokers: DSGResourceData[]
+): ServiceTopics[] {
+  messageBrokers.forEach((resource) => {
+    if (resource.resourceType === EnumResourceType.MessageBroker) {
+      throw new Error("resourceType is not EnumResourceType.MessageBroker");
+    }
+  });
```

Comment:
if you extract the data from the context in the function - you don't need this validation